font-types/src/pen.rs

```diff
@@ -85,3 +91,146 @@ impl PenCommand {
         }
     }
 }
+
+/// Pen adapter that outputs commands to SVG path data.
+///
+/// The target may be any type that implements `fmt::Write` such as
+/// `String`.
+pub struct SvgPen<T> {
+    target: T,
+    space: &'static str,
+}
+
+impl<T> SvgPen<T> {
+    pub fn new(target: T) -> Self {
+        Self { target, space: "" }
+    }
+
+    pub fn into_inner(self) -> T {
+        self.target
+    }
+}
+
+impl<T: fmt::Write> Pen for SvgPen<T> {
+    fn move_to(&mut self, x: f32, y: f32) {
+        let _ = write!(self.target, "{}M{},{}", self.space, x, y);
+        self.space = " ";
+    }
+
+    fn line_to(&mut self, x: f32, y: f32) {
+        let _ = write!(self.target, "{}L{},{}", self.space, x, y);
+        self.space = " ";
+    }
+
+    fn quad_to(&mut self, cx0: f32, cy0: f32, x: f32, y: f32) {
+        let _ = write!(self.target, "{}Q{},{} {},{}", self.space, cx0, cy0, x, y);
+        self.space = " ";
+    }
+
+    fn curve_to(&mut self, cx0: f32, cy0: f32, cx1: f32, cy1: f32, x: f32, y: f32) {
+        let _ = write!(
+            self.target,
+            "{}C{},{} {},{} {},{}",
+            self.space, cx0, cy0, cx1, cy1, x, y
+        );
+        self.space = " ";
+    }
+
+    fn close(&mut self) {
+        let _ = write!(self.target, "{}Z", self.space);
+        self.space = " ";
+    }
+}
+
+/// Pen adapter that collects commands into a target buffer.
+///
+/// The target may be any type that implements `Extend<PenCommand>` such as
+/// `Vec<PenCommand>`.
+pub struct BufferPen<T> {
```

Comment:
how about calling this RecordingPen like FontTools does?
Would be also nice if this had a "replay" method that takes another pen as argument and re-plays the recording onto that.

Reply:
That’s a better name, ty. Will change that and add a consuming replay method that is available when `T: IntoIterator<Item=PenCommand>`.

Reply:
I suggest also explicitly linking to recording pen and noting they do the same thing here and anywhere else we made a pen that matches, or is similar to, one fonttools ships.